internal/controller/standalone_pgadmin/controller.go

```diff
@@ -78,39 +80,29 @@ func ManagedReconciler(m ctrl.Manager) error {
 		Owns(&corev1.Service{}).
 		Watches(
 			v1beta1.NewPostgresCluster(),
-			handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, cluster client.Object) []ctrl.Request {
+			handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, cluster client.Object) []reconcile.Request {
 				return runtime.Requests(reconciler.findPGAdminsForPostgresCluster(ctx, cluster)...)
 			}),
 		).
 		Watches(
 			&corev1.Secret{},
-			handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, secret client.Object) []ctrl.Request {
+			handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, secret client.Object) []reconcile.Request {
 				return runtime.Requests(reconciler.findPGAdminsForSecret(ctx, client.ObjectKeyFromObject(secret))...)
 			}),
 		).
-		Complete(reconciler))
+		Complete(reconcile.AsReconciler(kubernetes, reconciler)))
```

Comment:
OK, so `AsReconciler` returns a reconciler from an ObjectReconciler, which `reconciler` is now because our `reconcile` func takes the object rather than a req.

Which is better because it offloads some generic logic from our reconcile to some k8s standard functions. (Also simplifies some tests where we were having to build a request.)

Did I get it? Am I missing something?

Reply:
👍🏻 That is all of it.